http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2ConnectionTerminatingException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty.internal.http2;
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+/**
+ * Exception indicating a connection is terminating
+ */
+@SdkInternalApi
+final class Http2ConnectionTerminatingException extends RuntimeException {
```

Comment:
Just a sanity check: we aren't relying on the existing exceptions we're replacing with this being an IOException (e.g. for retries) are we?

Reply:
Nope, this exception is only used internally to change the connection state to `CLOSE_TO_NEW` and is not returned to upstream.